### PR TITLE
Status events and Update to Cakephp5

### PR DESCRIPTION
**Pull request by jubbs**
I have updated the plugin to work with CakePHP5 and added the ability to listen for webhooks back from sendgrid and authenticate the webhooks with signed event tokens.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,13 +8,13 @@
...
 
-This branch is for use with CakePHP 4.0+. For CakePHP 3, please use cake-3.x branch.
+This branch is for use with CakePHP 5.0+. For CakePHP 4, please use cake-4.x branch.
 
 ## Requirements
 
 This plugin has the following requirements:
 
-* CakePHP 4.0 or greater.
+* CakePHP 5.0 or greater.
```
Comment (narendravaghela) [inline README.md:17]: You also need to update the php version to `>=8.1`

File: composer.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
     },
     "require": {
         "php": ">=7.2",
```
Comment (narendravaghela) [inline composer.json]: You also need to update php to `>=8.1`

**Comment by narendravaghela** [comment]
Hi @jubbs, thank you for your contribution. I have added some comments for small changes.

**Comment by jubbs** [comment]
I have made those changed.

**Comment by dereuromark** [comment]
Any update on this?
Would also be nice if this was then communicated via
https://github.com/FriendsOfCake/awesome-cakephp?tab=readme-ov-file#email

I just found it by chance.

**Comment by dereuromark** [comment]
This is for v7 right? as in v8 there are some API changes..
E.g.
new Client() doesnt work anymore, it requires 1st arg:
```php
	$host = gethostbyname($this->getConfig('apiEndpoint'));
	$this->Client = new Client($host);
```

**Comment by narendravaghela** [comment]
Hi @dereuromark
Thanks for the heads up. I will review the PR and merge it. Also, I will add it to the friends-of-cake/awesome-cakephp.

**Pull request merged**